lib/src/main/java/com/auth0/msg/AbstractMessage.java

```diff
@@ -0,0 +1,250 @@
+     * @param KeyJar the signing keyjar
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(KeyJar keyjar, Algorithm algorithm) throws
+            InvalidClaimsException, SerializationException {
+        header.put("alg", algorithm.getName());
+        header.put("typ", "JWT");
+        String signingKeyId = algorithm.getSigningKeyId();
+        if (signingKeyId != null) {
+            header.put("kid", signingKeyId);
+        }
+//        JWTCreator.Builder newBuilder = JWT.create().withHeader(this.header);
+//        for (Claim claimKey: claims.keySet()){
+//            newBuilder.withClaim(claimKey.name, (claimKey.type) claims.get(claimKey));
+//        }
+        return null;
+    }
+
+    /**
+     * Serialize the content of this instance (the claims map) into a jwt string
+     * @param Key the signing key
+     * @param String the algorithm to use in signing the JWT
+     * @return a jwt String
+     * @throws InvalidClaimsException
+     */
+    public String toJwt(Key key, Algorithm algorithm) throws InvalidClaimsException, SerializationException {
+        return null;
+    }
+
+    /**
+     * verify that the required claims are present
+     * @return whether the verification passed
+     */
+    protected boolean verify() {
+        //This method will set error if verification fails
+        List<String> errors = new ArrayList<String>();
+
+        List<String> reqClaims = getRequiredClaims();
+        if (!reqClaims.isEmpty() && this.claims.isEmpty()){
+            errors.add("The required claims are missing");
+            return false;
+        }
+
+        StringBuilder errorSB = new StringBuilder();
+        for (String req: reqClaims) {
+            if (!claims.containsKey(req)) {
+                errors.add("This message is missing required claim: " + req);
+                errorSB.append(req);
+            }
+        }
+
+        if (errorSB.length() != 0) {
+            errors.add("Message is missing required claims:" + errorSB.toString());
+            return false;
+        }
+
+        List<String> customClaims = new ArrayList<String>();
+        for(String claimName : claims.keySet()) {
+            // if knownClaim, validate claim
+            if (ClaimsValidator.isKnownClaim(claimName)) {
+                Boolean valid = ClaimsValidator.validate(claimName, claims.get(claimName), getMessageType());
+                if (!valid) {
+                    errors.add(claimName + "is an invalid claim");
+                }
+            } else {
+                customClaims.add(claimName);
+            }
+        }
+        if (!errors.isEmpty()) {
+            String aggregateError = "";
+            for (String err: errors){
+                aggregateError += err;
+            }
+            throw new InvalidClaimsException(aggregateError);
+        }
+        return false;
```

Comment:
Where do you set the class level variable 'verified'? and where do you set the class level Error?